### DISCUSSION TITLE
Extending the keycloak.v2 login theme in Keycloak 26

### DISCUSSION THREAD

**Discussion by senyahnoj** [keycloak/keycloak#33593] [Q&A]
With the new keycloak.v2 login theme release in Keycloak 26 today, how can we extend the theme?

Is there/will there be an NPM project for this like:

https://www.npmjs.com/package/@keycloak/keycloak-account-ui
https://www.npmjs.com/package/@keycloak/keycloak-admin-ui

?

**Comment by darius-m** [comment]
I think you can use the "classic" theme extension approach where you simply override the files you want from the base theme, and other than that use the base theme;  there's thankfully no need to recompile anything. The base theme should be here https://github.com/keycloak/keycloak/tree/main/themes/src/main/resources/theme/keycloak.v2/login

**Reply by darius-m** [reply]
You can use [the server development documentation](https://www.keycloak.org/docs/latest/server_development/index.html#_themes) about how to do it.

**Reply by senyahnoj** [reply]
Thanks for your reply!
